src/irc_bridge.py

```diff
@@ -32,44 +32,49 @@ def __init__(self, message: str):
         super().__init__(message)
 
 
-class IRCBridge(ops.Object):  # pylint: disable=too-few-public-methods
+class IRCBridgeObserver(ops.Object):  # pylint: disable=too-few-public-methods
     """A class representing the IRC bridge plugin for Synapse application.
 
     See https://github.com/matrix-org/matrix-appservice-irc/ for more details about it.
     """
 
-    def __init__(self, charm: ops.CharmBase, charm_state: CharmState):
+    def __init__(self, charm: CharmBaseWithState) -> None:
```

Comment:
I think the guidance is that charm state should be passed as an argument? And that we should also defer instantiation until the event hook going forward so it might not be needed at construct time anymore?

Reply:
answered below